build/cmake/lib/CMakeLists.txt

```diff
@@ -127,6 +127,7 @@ if (ZSTD_BUILD_STATIC)
     set_target_properties(
             libzstd_static
             PROPERTIES
+            POSITION_INDEPENDENT_CODE On
```

Comment:
You may wish a separate target for this. It's a tricky one, it's not uncommon that static libraries get linked into dynamic libraries, however sometimes fPIC does slightly damage performance.